From 94a1fe47f7a655cf3c02c196810ead608c54a3cc Mon Sep 17 00:00:00 2001
From: Melissa Wen <mwen@igalia.com>
Date: Tue, 24 Sep 2024 22:32:03 -0300
Subject: [PATCH 1/2] drm/amd/display: increase max surfaces aligned to planes
 and streams

091a97e542cf ("drm/amd/display: Fix warning. Set MAX_SURFACES to 3")
reduced the max number of surfaces since at that time, there was no use
for more. However, HW and driver evolves and there are now driver
versions that allow two overlay planes (max_slave_planes). Moreover,
commit 3cfd03b79425 ("drm/amd/display: update max streams per surface")
states 6 is the max surfaces supported asics can have. Therefore, update
MAX_SURFACES to match MAX_SURFACE_NUM, MAX_PLANES and MAX_STREAMS.

Link: https://gitlab.freedesktop.org/drm/amd/-/issues/3594
Signed-off-by: Melissa Wen <mwen@igalia.com>
---
 drivers/gpu/drm/amd/display/dc/dc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/display/dc/dc.h b/drivers/gpu/drm/amd/display/dc/dc.h
index 3992ad73165b..08b00b263533 100644
--- a/drivers/gpu/drm/amd/display/dc/dc.h
+++ b/drivers/gpu/drm/amd/display/dc/dc.h
@@ -57,7 +57,7 @@ struct dmub_notification;
 
 #define DC_VER "3.2.301"
 
-#define MAX_SURFACES 3
+#define MAX_SURFACES 6
 #define MAX_PLANES 6
 #define MAX_STREAMS 6
 #define MIN_VIEWPORT_SIZE 12
-- 
2.45.2

